packages/flame/lib/src/game/base_game.dart

```diff
@@ -238,6 +238,69 @@ class BaseGame extends Game with FPSCounter {
   /// show extra information on the screen when debug mode is activated
   bool debugMode = false;
 
+  /// Changes the priority of [component] and reorders the games component list.
+  ///
+  /// Returns true if changing the component's priority modified one of the
+  /// components that existed directly on the game and false if it
+  /// either was a child of another component, if it didn't exist at all or if
+  /// it was a component added directly on the game but its priority didn't
+  /// change.
+  bool changePriority(
+    Component component,
+    int priority, {
+    bool reorderRoot = true,
+  }) {
+    if (component.priority == priority) {
+      return false;
+    }
+    if (components.contains(component)) {
+      component.changePriorityWithoutResorting(priority);
+      if (reorderRoot) {
+        components.rebalanceAll();
+      }
+      return true;
+    } else if (component is BaseComponent) {
+      component.changePriorityWithoutResorting(priority);
+      BaseComponent? foundParent;
+      for (final parent in components.whereType<BaseComponent>()) {
+        if (parent == component.parent) {
+          foundParent = parent;
+          break;
+        }
+        parent.propagateToChildren<BaseComponent>((child) {
+          if (child == component.parent) {
+            foundParent = child;
+            return false;
+          }
+          return true;
+        });
+        if (foundParent != null) {
+          break;
+        }
+      }
```

Comment:
I think this could be extracted to a useful method on BaseComponent, what do you think? (findParent)

Reply:
Wait... Why am I doing this, `BaseComponent` already knows its parent, so for all of those we don't need to check and I'm missing to check for normal `Component` here. Maybe we should move the parent field to `Component` and this could be done a lot more efficiently. I'll re-do this part later today or tomorrow.

Reply:
Should be fixed now, please re-review :)